### PR TITLE
fix: exporting symbols from @bazel/concatjs

### PR DESCRIPTION
**Pull request by dymart**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

symbols from @bazel/typescript were not exported after they moved to @bazel/concatjs

## What is the new behavior?

export all needed symbols from @bazel/concatjs

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/concatjs/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "compilerOptions": {
+        "declaration": true,
+        "module": "umd",
+        "moduleResolution": "node",
+        "types": ["node"],
+        "target": "es5",
+        "downlevelIteration": true,
+        "rootDirs": [
+            ".",
+            "../../bazel-out/darwin-fastbuild/bin/packages/concatjs",
```
Comment (gregmagolan) [inline packages/concatjs/tsconfig.json:11]: CI error is that you're missing some possible paths here ```ERROR: /home/circleci/rules_nodejs/packages/concatjs/BUILD.bazel:23:11: Compiling TypeScript project //packages/concatjs:bazel_concatjs_lib [tsc -p packages/concatjs/tsconfig.json] failed: (Exit 2): tsc.sh failed: error executing command bazel-out/k8-opt-exec-2B5CBBC6/bin/external/npm/typescript/bin/tsc.sh --project packages/concatjs/tsconfig.json --outDir bazel-out/k8-opt-exec-2B5CBBC6/bin/packages/concatjs --rootDir packages/concatjs ... (remaining 3 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox
packages/concatjs/index.ts(131,15): error TS2307: Cannot find module './internal/tsc_wrapped' or its corresponding type declarations.```

Comment (gregmagolan) [reply packages/concatjs/tsconfig.json:11]:
```
k8-opt-exec-2B5CBBC6
darwin-opt-exec-2B5CBBC6
windows-opt-exec-2B5CBBC6
```
These are the outputs trees for the host (2B5CBBC6) execution platform

Comment (gregmagolan) [reply packages/concatjs/tsconfig.json:11]: I don't recall ever seeing a `linux-*` platform. Where did those ones come from?

**Comment by alexeagle** [comment]
The windows failure is due to lack of sandboxing. I can repro locally on linux as well:

```
bazel build --spawn_strategy=local //packages/concatjs/...

...
error TS5033: Could not write file '/home/alexeagle/.cache/bazel/_bazel_alexeagle/78c7479a1baf683797e1e6cd7cfefe83/execroot/build_bazel_rules_nodejs/bazel-out/k8-fastbuild/bin/packages/concatjs/web_test/test/stack_trace/test_folder/test.spec.js': EACCES: permission denied, open '/home/alexeagle/.cache/bazel/_bazel_alexeagle/78c7479a1baf683797e1e6cd7cfefe83/execroot/build_bazel_rules_nodejs/bazel-out/k8-fastbuild/bin/packages/concatjs/web_test/test/stack_trace/test_folder/test.spec.js'.
```

The reason is that `tsc` will include all the files from subdirectories in the program. Without sandboxing, that means it discovers files which weren't included in the `srcs`. thus `//packages/concatjs:bazel_concatjs_lib` will include all the ts files under `//packages/concatjs/internal:tsc_wrapped` and produce duplicate outputs for them.

**Comment by alexeagle** [comment]
Okay so you can add `includes` or `excludes` to the packages/concatjs/tsconfig.json as the TS-idiomatic way to prevent those files from nested directories getting included in the program.

HOWEVER then you hit the next TS bug https://github.com/microsoft/TypeScript/issues/22208 which means that when it sees `export * from './internal/tsc_wrapped'` outside of a sandbox, it will look around the filesystem and add internal/tsc_wrapped/*.ts to the program, no matter how much we try to get `bazel-out/internal/tsc_wrapped/*.d.ts` to be available to the compiler. Then it will still try to emit on top of existing files.

**Comment by dymart** [comment]
import with this: @bazel/concatjs/internal/tsc_wrapped

**Comment by devversion** [comment]
@dymart Haha, I was going to ask if you plan on still doing this somewhere else, but I agree that the internal import is sufficient. thx for the update here (I was having some todos referring to this PR)

**Pull request closed without merging**